### PR TITLE
Pa survey refactor

### PR DESCRIPTION
**Pull request by jbrou015**

### CODE REVIEW

File: ComplaintFormCore/ClientApp/src/pa/newPaSurvey.ts [hunk 1]
```diff
@@ -137,4 +145,34 @@ export class NewPaSurvey extends SurveyBase {
...
+        }
+
+        if (surveyObj.data["FilingComplaintOnOwnBehalf"] === "someone_else") {
+            const liNode = ul_documentation_info.querySelector(".filingComplaintOnOwnBehalf");
+            if (liNode != null) {
+                liNode.classList.remove("sv-hidden");
+            }
+        }
+
+        if (surveyObj.data["NatureOfComplaint"].filter(x => x === "NatureOfComplaintDenialOfAccess").length > 0) {
```
Comment (jbrou015) [inline ComplaintFormCore/ClientApp/src/pa/newPaSurvey.ts:171]: FYI I am trying to set this up with **calculatedValues**

**Review by joshcampbell191 — CHANGES_REQUESTED** [review]

File: ComplaintFormCore/ClientApp/src/pa/newPaSurvey.ts [hunk 2]
```diff
@@ -128,7 +127,10 @@ export class NewPaSurvey extends SurveyBase {
                 "success",
                 options.files.map(file => ({
                     file: file,
-                    content: responseData[file.name]
+                    content: `/api/File/Get?complaintId=${this.authToken}&fileUniqueId=${
+                        responseData[file.name].content as string
+                    }&filename=${file.name as string}`,
+                    size: responseData[file.name].size
```
[the remark this answers is not in the document]
Comment (jbrou015) [reply ComplaintFormCore/ClientApp/src/pa/newPaSurvey.ts:169]: It is the link associated to each file so when the file is clicked it knows where to fetch the file data. It is needed.

File: ComplaintFormCore/ClientApp/src/survey.ts [hunk 3]
```diff
@@ -32,20 +34,12 @@ export abstract class SurveyBase extends Survey {
         this.registerWidgets();
         this.registerEventHandlers();
         this.registerCustomProperties();
+
+        this.storageName = storageName;
```
[the remark this answers is not in the document]
Comment (jbrou015) [reply ComplaintFormCore/ClientApp/src/survey.ts]: Looks like you've done it already!

File: ComplaintFormCore/ClientApp/src/widgets/filemeterwidget.ts [hunk 4]
```diff
@@ -155,6 +196,18 @@ export class FileMeterWidget extends Widget {
...
+                if (file.size) {
+                    total += file.size;
+                }
+            });
+
+            return total;
+            // return files.reduce((previous, current) => previous + current.size, 0);
+        } else {
+            return 0;
+        }
```
[the remark this answers is not in the document]
Comment (jbrou015) [reply ComplaintFormCore/ClientApp/src/widgets/filemeterwidget.ts:211]: Ok. The original code was creating a bug but it is now fixed.

File: ComplaintFormCore/wwwroot/sample-data/survey_pa_complaint.json [hunk 5]
```diff
@@ -1305,6 +1283,21 @@
...
+    {
+      "name": "DocumentationInfoFilingComplaintOnOwnBehalfGetClass",
+      "expression": "iif({FilingComplaintOnOwnBehalf} = 'someone_else', 'mrgn-bttm-sm', 'mrgn-bttm-sm sv-hidden')",
+      "includeIntoResult": true
+    },
+    {
+      "name": "DocumentationInfoNatureOfComplaintDenialOfAccessGetClass",
+      "expression": "iif({NatureOfComplaint} contains 'NatureOfComplaintDenialOfAccess', 'mrgn-bttm-sm', 'mrgn-bttm-sm sv-hidden')",
+      "includeIntoResult": true
     }
```
[the remark this answers is not in the document]
Comment (jbrou015) [reply ComplaintFormCore/wwwroot/sample-data/survey_pa_complaint.json:1301]: Ouch! This is a <ul> containing 5 <li> and 2 of them are always shown. If you want to do it this way we will need 6 different html elements not 2.

File: ComplaintFormCore/ClientApp/src/survey.ts [hunk 6]
```diff
@@ -178,6 +260,52 @@ export abstract class SurveyBase extends Survey {
...
+        }
+    }
+
+    private handleOnGetQuestionTitle(sender: SurveyModel, options: any): void {
+        //  This is to add * at the beginning of a required question. The property requiredText
+        //  is set as 'required' later in the code
+        if (options.question.isRequired) {
+            options.title = `<span class='sv_q_required_text'>&ast;&nbsp;</span>${options.title as string}`;
+        }
+    }
```
[the remark this answers is not in the document]
Comment (joshcampbell191) [reply ComplaintFormCore/ClientApp/src/survey.ts]: This was added to the survey.css

File: ComplaintFormCore/ClientApp/src/models/localStorage.ts [hunk 7]
```diff
@@ -0,0 +1,4 @@
+﻿export interface LocalStorage {
```
[the remark this answers is not in the document]
Comment (joshcampbell191) [reply ComplaintFormCore/ClientApp/src/models/localStorage.ts]: This was already changed.

File: ComplaintFormCore/ClientApp/src/survey.ts [hunk 8]
```diff
@@ -178,6 +260,52 @@ export abstract class SurveyBase extends Survey {
         this.displayErrorSummary(questionErrors);
     }
 
+    private handleOnUpdateQuestionCssClasses(sender: SurveyModel, options: any): void {
+        const classes = options.cssClasses;
+
+        //  If it is the preview mode...
+        if (sender.isDisplayMode === true) {
```
[the remark this answers is not in the document]
Comment (joshcampbell191) [reply ComplaintFormCore/ClientApp/src/survey.ts]: No longer relevant.

File: ComplaintFormCore/ClientApp/src/survey.ts [hunk 9]
```diff
@@ -178,6 +260,52 @@ export abstract class SurveyBase extends Survey {
...
+        //  This is to add * at the beginning of a required question. The property requiredText
+        //  is set as 'required' later in the code
+        if (options.question.isRequired) {
+            options.title = `<span class='sv_q_required_text'>&ast;&nbsp;</span>${options.title as string}`;
+        }
+    }
+
+    private handleOnAfterRenderPage(sender: SurveyModel, options: any): void {
+        window.document.title = options.page.title;
+    }
```
[the remark this answers is not in the document]
Comment (joshcampbell191) [reply ComplaintFormCore/ClientApp/src/survey.ts]: No longer relevant.

File: ComplaintFormCore/ClientApp/src/survey.ts [hunk 10]
```diff
@@ -178,6 +260,52 @@ export abstract class SurveyBase extends Survey {
...
+            //  Panels that contains information html for example.
+            //  It also hides the 'pages'! The reason is because on preview, the pages become panels
+            //  and therefore going thru this code.
+            classes.panel.container = "sv-hidden";
+        } else {
+            //  This is a class found in GoC and was used in the original project.
+            //  It adds a border around a panel and a different backgroud color
+            classes.panel.container += " well";
+        }
+    }
```
[the remark this answers is not in the document]
Comment (joshcampbell191) [reply ComplaintFormCore/ClientApp/src/survey.ts]: No longer relevant.

File: ComplaintFormCore/wwwroot/sample-data/survey_pa_complaint.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (joshcampbell191) [reply ComplaintFormCore/wwwroot/sample-data/survey_pa_complaint.json:1301]: Let me check and see if there's anything else we can do for this.

File: ComplaintFormCore/ClientApp/src/pa/paSurvey.ts [hunk 11]
```diff
@@ -1,12 +1,12 @@
 import Vue from "vue";
 import * as Survey from "survey-vue";
-import { saveStateLocally, storageName_PA, loadStateLocally } from "../surveyLocalStorage";
 import { initSurvey, initSurveyModelEvents, initSurveyModelProperties } from "../surveyInit";
 import { initSurveyFile, initSurveyFileModelEvents } from "../surveyFile";
 import { printProblemDetails, getTranslation } from "../surveyHelper";
 import * as SurveyNavigation from "../surveyNavigation";
 import * as Ladda from "ladda";
 import { paTestData } from "./pa_test_data";
+import { SurveyLocalStorage } from "../surveyLocalStorage";
```
[the remark this answers is not in the document]
Comment (jbrou015) [reply ComplaintFormCore/ClientApp/src/pa/paSurvey.ts:9]: Yes but we're gonna get rid of paSurvey.ts soon

File: ComplaintFormCore/ClientApp/src/pa/newPaSurvey.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (joshcampbell191) [reply ComplaintFormCore/ClientApp/src/pa/newPaSurvey.ts:169]: Is the `size` property still required?

**Review by joshcampbell191 — APPROVED** [review]

**Pull request merged**